src/System.Private.CoreLib/src/System/Threading/Tasks/Task.CoreCLR.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Tracing;
+
+namespace System.Threading.Tasks
+{
+    public partial class Task : IAsyncResult, IDisposable
+    {
+        static partial void EtwNewID(int newId)
+        {
+            TplEtwProvider.Log.NewID(newId);
+        }
+
+        partial void EtwTaskStarted(Task previousTask, ref Guid savedActivityID)
```

Comment:
I'm a little concerned that this factoring is going to impact performance, as most of these methods are not going to get inlined automatically, and we're now reading the static log multiple times where previously it was read only once.

Reply:
I'd expect the read from the static read-only field to be inlined. If you are concern about extra call maybe aggressive inlining would help but not sure if the call price is even measurable in this context.

Reply:
It usually is inlined and eliminated, but it depends on the compilation tier. Explicit aggressive inlining or aggressive optimization may actually hurt the performance. I think it has to do with the fact that a static read-only field may not be assigned by the time the first JIT compilation is performed, but if you let tiering do its thing then later recompilation will see it assigned and optimize properly. Whatever the reason I suggest not to do it without a benchmark backing it up.

Reply:
This split should not be needed. It should be fine to have same TplEtwProvider  as is in CoreCLR. TplEtwProvider is in CoreRT already; it just factored differently for historic reasons.

Reply:
https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/Internal/Threading/Tasks/Tracing/TaskTrace.cs and https://github.com/dotnet/corert/blob/master/src/System.Private.Threading/src/Internal/Threading/TaskTraceCallbacksImplementation.cs can be deleted in CoreRT; and https://github.com/dotnet/corert/blob/master/src/System.Private.Threading/src/System/Threading/Tasks/TPLETWProvider.cs can be moved to CoreLib and unified with the CoreCLR one.

Reply:
@jkotas done (will make TplEtwProvider changes in different PR). I kept the Etw separation for easier reading and I don't think it should cause any perf issue.